src/components/Home.js

```diff
@@ -30,77 +51,60 @@ export class Home extends React.Component {
     this.handleBreaksToSpaces = this.handleBreaksToSpaces.bind(this)
     this.handleRemoveWhiteSpaces = this.handleRemoveWhiteSpaces.bind(this)
     this.handleSwap = this.handleSwap.bind(this)
-    this.state = {
-      diffArray: [],
-      result: false,
-      identical: false,
-      swapped: false,
-      original: '',
-      changed: ''
-    }
   }
 
-  handleCompare () {
-    const original = this.state.original
-    const changed = this.state.changed
-    const diffArray = Diff.diffChars(original, changed)
+  handleCompare (original, changed) {
     if (original === changed) {
-      this.setState({ identical: true, result: false })
+      store.dispatch(identical())
```

Comment:
1. One event handler -> one action
2. No implicit dependencies between actions
3. If one action triggers another action, then thunk must be used